### PR TITLE
Feature/teaser linked

### PR DESCRIPTION
**Pull request by DavithkbY**
Finished linked teaser component

### CODE REVIEW

File: src/data/contentful/teaser-linked/ctf-teaser-linked-gql.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+export function CtfTeaserLinkedGql(props: DefaultQueryParameters) {
+  const data = useCtfComponentTeaserLinkedQuery({
+    id: props.id,
+    locale: props.locale,
+    preview: props.preview,
+  });
+
+  if (data.data?.teaserLinked) {
+    return <TeaserLinked data={data} />
```
Comment (ceneva) [inline src/data/contentful/teaser-linked/ctf-teaser-linked-gql.tsx:17]: Only data should be returned in the data layer. Please use the mapping to link data to components.

File: src/mapping/contentful/teaser-linked.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    description: description,
+    descriptionClasses: descriptionContext.context_intro
+  }
+
+  const buttonProps: ButtonProps = {
+    buttonText: teaserLinked.buttonText!,
+    buttonUrl: slug,
+  }
+
+  return (
```
Comment (ceneva) [inline src/mapping/contentful/teaser-linked.tsx:40]: Can you change this into a component in the component folder and pass the data? Flatten the data for the component, and give it a generic name, it doesn't matter for the component layer where the background-image comes from.

File: src/mapping/sanity/reference-client.tsx [hunk 3]
```diff
@@ -1,4 +1,30 @@
...
-// remove data source from component
-export {}
+import { ReferenceClients } from '@src/components/features/decoupled-components/reference-clients/reference-clients';
+import {
+  ReferenceClient,
+  ReferenceClientsProps,
+} from '@src/components/features/decoupled-components/reference-clients/reference-clients.typings';
+import { TeaserItemSanity } from '@src/mapping/sanity/solution';
+
+interface ReferenceClientProps {
```
Comment (ceneva) [inline src/mapping/sanity/reference-client.tsx:8]: Put typings in separate file and group them where they belong, either in de component folder or the data folder. Mapping should only link existing structures, not define them.

File: src/data/contentful/teaser-linked/ctf-teaser-linked-gql.tsx (hunk 1 above)
Comment (DavithkbY) [reply src/data/contentful/teaser-linked/ctf-teaser-linked-gql.tsx:17]: It is linking the data to the TeaserLinked component in the mapping folder.
import { TeaserLinked } from '@src/mapping/contentful/teaser-linked';

File: src/mapping/contentful/mappings.ts [hunk 4]
```diff
@@ -1,4 +1,5 @@
 import dynamic from 'next/dynamic';
+import { CtfTeaserLinkedGql } from '@src/data/contentful/teaser-linked/ctf-teaser-linked-gql';
```
Comment (Polleke007) [inline src/mapping/contentful/mappings.ts:2]: Are you sure this is correct, seems like all the other imports for gql are not requried

File: src/mapping/contentful/teaser-linked.tsx (hunk 2 above)
Comment (Polleke007) [reply src/mapping/contentful/teaser-linked.tsx:40]: This is already done and merged in this branch

**Pull request merged**